src/main/java/com/redhat/devtools/intellij/lsp4ij/operations/completion/LSPCompletionContributor.java

```diff
@@ -54,32 +54,46 @@ public void fillCompletionVariants(@NotNull CompletionParameters parameters, @No
         PsiFile file = parameters.getOriginalFile();
         Project project = file.getProject();
         int offset = parameters.getOffset();
-        CompletableFuture<List<Pair<LanguageServerWrapper, LanguageServer>>> completionLanguageServersFuture = initiateLanguageServers(project, document);
+
+        ProgressManager.checkCanceled();
+
+        final CancellationSupport cancellationSupport = new CancellationSupport();
         try {
+            CompletableFuture<List<LanguageServerItem>> completionLanguageServersFuture = initiateLanguageServers(project, document);
+            cancellationSupport.execute(completionLanguageServersFuture);
+            ProgressManager.checkCanceled();
+
             /*
              process the responses out of the completable loop as it may cause deadlock if user is typing
              more characters as toProposals will require as read lock that this thread already have and
              async processing is occuring on a separate thread.
              */
+            ProgressManager.checkCanceled();
```

Comment:
there's already a `checkCanceled()` call right before this one

Reply:
fixed